tool_watershed/watershed_analysis_networkx.py

```diff
@@ -235,10 +242,18 @@ def cells_as_multipolygon(self, cell_ids: set):
 
         WARNING! May result in memory error if somehwat large set of cells is given as input
         """
-        upstream_cells = self._gr.cells.filter(id__in=list(cell_ids))
+        # upstream_cells = self._gr.cells.filter(id__in=list(cell_ids))
         cell_drv = ogr.GetDriverByName("MEMORY")
         cell_ds = cell_drv.CreateDataSource("")
-        threedigrid_to_ogr(threedigrid_src=upstream_cells, tgt_ds=cell_ds, attributes={}, attr_data_types={})
+        result = self.model.get_result(self.result_id)
+        gridadmin_gpkg = result.parent().path.with_suffix('.gpkg')
+        threedigrid_to_ogr(
+            tgt_ds=cell_ds,
+            layer_name="cell",
+            gridadmin_gpkg=gridadmin_gpkg,
```

Comment:
This look unnecessarily complex to me. I would prefer ``gridadmin_gpkg=self.gridadmin.gpkg``, then give ``__init__()`` an argument ``gridadmin_gpkg: Union[str, Path]`` (I don't think it should be optional, i.e. leave out `` = None``) and set ``self.gridadmin_gpkg = str(gridadmin_gpkg)``